### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by babogdanov**
### Backend
- Extend user entity with reset token field
- Migrate user module to auth module with login, registration and token invalidation endpoints
- Add JWT-based authentication and stored password encryption

### Frontend
- Setup react-query with axios, including generic hooks for type-safe queries and mutations
- Add login and register pages + generic form validation
- Add logic for storing, updating and clearing issued JWT tokens in local storage
- Add protected routes that may only be accessed by authenticated users
- Set up react-toastify for app-wide pop-up messaging

### Misc
- Normalize linting and formatting rules across FE and BE

### CODE REVIEW

File: .vscode/settings.json
Comment (ditachev) [inline .vscode/settings.json:1]: Missing newline at the end

File: backend/.eslintrc.js [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
   plugins: ['@typescript-eslint/eslint-plugin'],
-  extends: [
-    'plugin:@typescript-eslint/recommended',
-    'plugin:prettier/recommended',
-  ],
   root: true,
   env: {
     node: true,
-    jest: true,
+    jest: true
```
Comment (ditachev) [inline backend/.eslintrc.js]:
```suggestion
    jest: true,
```

File: backend/.prettierrc
Comment (ditachev) [inline backend/.prettierrc:1]: Missing newline at the end

File: backend/src/api/auth/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  @Post('register')
+  async register(@Body() registerDto: RegisterDto) {
+    const doesUserExist = await this.authService.getUser({
+      email: registerDto.email,
+    })
+
+    if (doesUserExist) {
+      throw new ConflictException('User already exists.')
+    }
```
Comment (ditachev) [inline backend/src/api/auth/auth.controller.ts]:
```suggestion
    const userExists = await this.authService.getUser({
      email: registerDto.email,
    })

    if (userExists) {
      throw new ConflictException('User already exists.')
    }
```

File: backend/src/api/auth/dto/register.dto.ts
Comment (ditachev) [inline backend/src/api/auth/dto/register.dto.ts:1]: This file is nearly identical to `email.dto.ts`. Can we somehow unify them?

File: frontend/src/AppWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import './index.css'
+// Load react toastify CSS
+import 'react-toastify/dist/ReactToastify.css'
+import App from './App'
+import { queryClient } from './api/queryClient'
+// Create a client
+
+const AppWrapper = () => {
+  return (
```
Comment (ditachev) [inline frontend/src/AppWrapper.tsx]:
```suggestion
import { queryClient } from './api/queryClient'

// Create a client
const AppWrapper = () => {
  return (
```

**Review by ditachev — APPROVED** [review]
Minor stylistic comments

File: frontend/README.md [hunk 4]
```diff
@@ -1,3 +1,3 @@
 ## How to run
 
-`yarn && yarn dev`
\ No newline at end of file
+`yarn && yarn dev`
```
Comment (xrebell) [inline frontend/README.md]: Shouldn't this be `yarn && yarn start`?

**Review by xrebell — APPROVED** [review]
LGTM overall.

**Pull request merged**